src/loc/LocPublishButton.tsx

```diff
@@ -1,32 +1,37 @@
 import { Fees } from "@logion/node-api";
 import Button from "../common/Button";
-import { useState, useEffect } from "react";
+import { useState, useEffect, useMemo, useCallback } from "react";
 import ProcessStep from "../common/ProcessStep";
-import Alert from "../common/Alert";
 import { PublishProps, PublishState, PublishStatus, LinkItem, FileItem, MetadataItem } from "./LocItem";
 import Icon from "../common/Icon";
 import { useLocContext } from "./LocContext";
-import ClientExtrinsicSubmitter, { Call, CallCallback } from "src/ClientExtrinsicSubmitter";
 import LocPublicDataDetails from "./LocPublicDataDetails";
 import LocPrivateFileDetails from "./LocPrivateFileDetails";
 import LocLinkDetails from "./LocLinkDetails";
 import LocItemEstimatedFees from "./LocItemEstimatedFees";
+import { CallCallback, useLogionChain } from "src/logion-chain";
+import ExtrinsicSubmissionStateView from "src/ExtrinsicSubmissionStateView";
 
 export default function LocPublishButton(props: PublishProps) {
     const [ publishState, setPublishState ] = useState<PublishState>({ status: PublishStatus.NONE });
-    const [ call, setCall ] = useState<Call>();
     const { mutateLocState, loc } = useLocContext();
     const [ fees, setFees ] = useState<Fees | undefined | null>();
+    const { submitCall, clearSubmissionState } = useLogionChain();
 
-    useEffect(() => {
-        if (publishState.status === PublishStatus.PUBLISH_PENDING) {
-            setPublishState({ status: PublishStatus.PUBLISHING });
-            const call: Call = async (callback: CallCallback) =>
-                mutateLocState(async current =>
-                    props.publishMutator(current, callback));
-            setCall(() => call);
+    const call = useMemo(() => {
+        return async (callback: CallCallback) =>
+            mutateLocState(async current =>
+                props.publishMutator(current, callback));
+    }, [ mutateLocState, props ]);
+
+    const publish = useCallback(async () => {
+        setPublishState({ status: PublishStatus.PUBLISHING });
+        try {
+            await submitCall(call);
+        } finally {
+            setPublishState({ status: PublishStatus.DONE });
         }
```

Comment:
Sounds weird to do nothing if an exception is thrown, although this may make sense in this particular case. Maybe just `console.error()` ?

Reply:
Actually, it is not. The state only depends on the call to be executed, whatever the outcome. The error is displayed by ExtrinsicSubmissionStateView.